Objects/dictobject.c

```diff
@@ -3660,6 +3953,17 @@ _PyDict_SizeOf(PyDictObject *mp)
     return (Py_ssize_t)res;
 }
 
+Py_ssize_t
+_PyDict_SizeOf(PyDictObject *mp)
```

Comment:
I think we'll want to do the locking in the callers of `_PyDict_SizeOf`, but I'd also prefer that we don't change all the callers to `_PyDict_SizeOf` in this PR.

Reply:
Okay, I'll leave it here for now and we can push it up later.